plugins/module_utils/foreman_helper.py

```diff
@@ -655,10 +690,237 @@ def exit_json(self, changed=False, **kwargs):
         super(ForemanAnsibleModule, self).exit_json(**kwargs)
 
 
-class ForemanEntityAnsibleModule(ForemanAnsibleModule):
+class ResolverEntityMixin(object):
+    """ This mixin add automatic entities resolution based on provided attributes/ sub entities options.
+        It add following options to entity_spec 'entity' and 'entity_list' types:
```

Comment:
I think this should be added to https://github.com/theforeman/foreman-ansible-modules/blob/master/docs/developing.md#specification-of-the-entity_spec :)